src/cpp/src/scheduler.hpp

```diff
@@ -237,6 +255,9 @@ class Scheduler {
                 size_t available_slots = currently_allocated_token_slots - occupied_token_slots,
                        required_slots = num_scheduled_tokens > available_slots ? num_scheduled_tokens - available_slots : 0;
                 size_t num_required_blocks = (required_slots + block_size - 1) / block_size, num_free_blocks = m_block_manager.num_free_blocks();
+                if (num_free_blocks == 0) {
```

Comment:
```suggestion
                if (num_required_blocks > num_free_blocks ) {
```
should not it be like this?

Reply:
Yes, corrected.